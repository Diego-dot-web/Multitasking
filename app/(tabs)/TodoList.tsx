import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  TextInput,
  TouchableOpacity,
  Alert,
} from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { useTheme } from "../../context/ThemeContext";
import { colors } from "@/components/Colors";
import { Card } from "@/components/Card";
import { Checkbox } from "@/components/Checkbox";
import { PlusCircle, Trash2, Filter, SortDesc } from "@/components/Icons";

interface Todo {
  id: number;
  text: string;
  completed: boolean;
  priority: "low" | "medium" | "high";
  createdAt: number;
}

export default function TodoList() {
  const { isDark } = useTheme();
  const theme = isDark ? colors.dark : colors.light;

  const [todos, setTodos] = useState<Todo[]>([]);
  const [newTodo, setNewTodo] = useState("");
  const [filter, setFilter] = useState<"all" | "active" | "completed">("all");
  const [sort, setSort] = useState<"newest" | "oldest" | "priority">("newest");
  const [newTodoPriority, setNewTodoPriority] = useState<
    "low" | "medium" | "high"
  >("medium");

  // Cargar tareas desde AsyncStorage
  useEffect(() => {
    const loadTodos = async () => {
      try {
        const savedTodos = await AsyncStorage.getItem("todos");
        if (savedTodos) {
          setTodos(JSON.parse(savedTodos));
        }
      } catch (e) {
        console.error("Error loading todos", e);
      }
    };

    loadTodos();
  }, []);

  // Guardar tareas en AsyncStorage cuando cambien
  useEffect(() => {
    const saveTodos = async () => {
      try {
        await AsyncStorage.setItem("todos", JSON.stringify(todos));
      } catch (e) {
        console.error("Error saving todos", e);
      }
    };

    saveTodos();
  }, [todos]);

  const addTodo = () => {
    if (newTodo.trim() !== "") {
      const newTodoItem: Todo = {
        id: Date.now(),
        text: newTodo,
        completed: false,
        priority: newTodoPriority,
        createdAt: Date.now(),
      };
      setTodos([...todos, newTodoItem]);
      setNewTodo("");
      setNewTodoPriority("medium");
    }
  };

  const toggleTodo = (id: number) => {
    setTodos(
      todos.map((todo) =>
        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
      ),
    );
  };

  const deleteTodo = (id: number) => {
    Alert.alert(
      "Eliminar misión",
      "¿Estás seguro de que quieres eliminar esta misión?",
      [
        { text: "Cancelar", style: "cancel" },
        {
          text: "Eliminar",
          onPress: () => {
            setTodos(todos.filter((todo) => todo.id !== id));
          },
          style: "destructive",
        },
      ],
    );
  };

  const showFilterMenu = () => {
    Alert.alert("Filtrar misiones", "Selecciona un filtro", [
      { text: "Todas", onPress: () => setFilter("all") },
      { text: "Activas", onPress: () => setFilter("active") },
      { text: "Completadas", onPress: () => setFilter("completed") },
      { text: "Cancelar", style: "cancel" },
    ]);
  };

  const showSortMenu = () => {
    Alert.alert("Ordenar misiones", "Selecciona un orden", [
      { text: "Más recientes", onPress: () => setSort("newest") },
      { text: "Más antiguas", onPress: () => setSort("oldest") },
      { text: "Por prioridad", onPress: () => setSort("priority") },
      { text: "Cancelar", style: "cancel" },
    ]);
  };

  const showPriorityMenu = () => {
    Alert.alert(
      "Seleccionar prioridad",
      "Elige la prioridad para la nueva misión",
      [
        { text: "Alta", onPress: () => setNewTodoPriority("high") },
        { text: "Media", onPress: () => setNewTodoPriority("medium") },
        { text: "Baja", onPress: () => setNewTodoPriority("low") },
        { text: "Cancelar", style: "cancel" },
      ],
    );
  };

  const getPriorityColor = (priority: "low" | "medium" | "high") => {
    switch (priority) {
      case "high":
        return colors.red;
      case "medium":
        return colors.yellow;
      case "low":
        return colors.green;
    }
  };

  const filteredTodos = todos.filter((todo) => {
    if (filter === "all") return true;
    if (filter === "active") return !todo.completed;
    if (filter === "completed") return todo.completed;
    return true;
  });

  const sortedTodos = [...filteredTodos].sort((a, b) => {
    if (sort === "newest") return b.createdAt - a.createdAt;
    if (sort === "oldest") return a.createdAt - b.createdAt;
    if (sort === "priority") {
      const priorityOrder = { high: 0, medium: 1, low: 2 };
      return priorityOrder[a.priority] - priorityOrder[b.priority];
    }
    return 0;
  });

  const getCompletionStats = () => {
    const total = todos.length;
    const completed = todos.filter((todo) => todo.completed).length;
    const percentage = total > 0 ? Math.round((completed / total) * 100) : 0;
    return { total, completed, percentage };
  };

  const stats = getCompletionStats();

  return (
    <View style={[styles.container, { backgroundColor: theme.background }]}>
      <View style={styles.header}>
        <Text style={[styles.title, { color: theme.text }]}>Tus Misiones</Text>
        <Text style={[styles.subtitle, { color: theme.muted }]}>
          Gestiona tus tareas y conquista tus objetivos.
        </Text>
      </View>

      <Card style={styles.todoCard}>
        <View style={styles.todoHeader}>
          <View>
            <Text style={[styles.todoTitle, { color: theme.text }]}>
              Lista de Misiones
            </Text>
            <Text style={[styles.todoSubtitle, { color: theme.muted }]}>
              {stats.completed} de {stats.total} misiones completadas (
              {stats.percentage}%)
            </Text>
          </View>

          <View style={styles.headerButtons}>
            <TouchableOpacity
              style={[styles.iconButton, { borderColor: theme.border }]}
              onPress={showFilterMenu}
            >
              <Filter width={16} height={16} color={theme.muted} />
            </TouchableOpacity>

            <TouchableOpacity
              style={[styles.iconButton, { borderColor: theme.border }]}
              onPress={showSortMenu}
            >
              <SortDesc width={16} height={16} color={theme.muted} />
            </TouchableOpacity>
          </View>
        </View>

        <View style={styles.inputContainer}>
          <View style={styles.inputWrapper}>
            <TextInput
              style={[
                styles.input,
                {
                  color: theme.text,
                  borderColor: theme.border,
                  backgroundColor: isDark ? theme.card : theme.background,
                },
              ]}
              placeholder="Añadir nueva misión..."
              placeholderTextColor={theme.muted}
              value={newTodo}
              onChangeText={setNewTodo}
              onSubmitEditing={addTodo}
            />

            <TouchableOpacity
              style={[
                styles.priorityButton,
                { backgroundColor: `${getPriorityColor(newTodoPriority)}20` },
              ]}
              onPress={showPriorityMenu}
            >
              <View
                style={[
                  styles.priorityDot,
                  { backgroundColor: getPriorityColor(newTodoPriority) },
                ]}
              />
            </TouchableOpacity>
          </View>

          <TouchableOpacity
            style={[styles.addButton, { backgroundColor: colors.primary }]}
            onPress={addTodo}
          >
            <PlusCircle width={20} height={20} color="#fff" />
            <Text style={styles.addButtonText}>Añadir</Text>
          </TouchableOpacity>
        </View>

        <ScrollView style={[styles.todoList, { borderColor: theme.border }]}>
          {sortedTodos.length > 0 ? (
            sortedTodos.map((todo) => (
              <View
                key={todo.id}
                style={[
                  styles.todoItem,
                  {
                    backgroundColor: todo.completed
                      ? `${theme.muted}10`
                      : "transparent",
                    borderBottomColor: theme.border,
                  },
                ]}
              >
                <Checkbox
                  checked={todo.completed}
                  onToggle={() => toggleTodo(todo.id)}
                />

                <View style={styles.todoContent}>
                  <View style={styles.todoTextContainer}>
                    <View
                      style={[
                        styles.priorityIndicator,
                        { backgroundColor: getPriorityColor(todo.priority) },
                      ]}
                    />
                    <Text
                      style={[
                        styles.todoText,
                        {
                          color: theme.text,
                          textDecorationLine: todo.completed
                            ? "line-through"
                            : "none",
                          opacity: todo.completed ? 0.6 : 1,
                        },
                      ]}
                    >
                      {todo.text}
                    </Text>
                  </View>

                  <TouchableOpacity
                    style={styles.deleteButton}
                    onPress={() => deleteTodo(todo.id)}
                  >
                    <Trash2 width={16} height={16} color={theme.muted} />
                  </TouchableOpacity>
                </View>
              </View>
            ))
          ) : (
            <View style={styles.emptyState}>
              <Text style={[styles.emptyText, { color: theme.muted }]}>
                {filter !== "all"
                  ? `No hay misiones ${filter === "active" ? "activas" : "completadas"}.`
                  : "¡Añade una nueva misión para comenzar tu aventura!"}
              </Text>

              {filter !== "all" && (
                <TouchableOpacity
                  style={[styles.emptyButton, { borderColor: theme.border }]}
                  onPress={() => setFilter("all")}
                >
                  <Text style={[styles.emptyButtonText, { color: theme.text }]}>
                    Ver todas las misiones
                  </Text>
                </TouchableOpacity>
              )}
            </View>
          )}
        </ScrollView>
      </Card>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
  },
  header: {
    marginBottom: 16,
  },
  title: {
    fontSize: 24,
    fontWeight: "bold",
    marginBottom: 4,
  },
  subtitle: {
    fontSize: 16,
  },
  todoCard: {
    flex: 1,
  },
  todoHeader: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginBottom: 16,
  },
  todoTitle: {
    fontSize: 18,
    fontWeight: "bold",
  },
  todoSubtitle: {
    fontSize: 14,
  },
  headerButtons: {
    flexDirection: "row",
    gap: 8,
  },
  iconButton: {
    width: 32,
    height: 32,
    borderRadius: 4,
    borderWidth: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  inputContainer: {
    flexDirection: "row",
    gap: 8,
    marginBottom: 16,
  },
  inputWrapper: {
    flex: 1,
    flexDirection: "row",
  },
  input: {
    flex: 1,
    height: 44,
    borderWidth: 1,
    borderRadius: 4,
    paddingHorizontal: 12,
  },
  priorityButton: {
    width: 44,
    height: 44,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 4,
  },
  priorityDot: {
    width: 12,
    height: 12,
    borderRadius: 6,
  },
  addButton: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: 16,
    height: 44,
    borderRadius: 4,
  },
  addButtonText: {
    color: "#fff",
    fontWeight: "500",
    marginLeft: 8,
  },
  todoList: {
    flex: 1,
    borderWidth: 1,
    borderRadius: 4,
  },
  todoItem: {
    flexDirection: "row",
    alignItems: "center",
    padding: 12,
    borderBottomWidth: 1,
  },
  todoContent: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginLeft: 12,
  },
  todoTextContainer: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
  },
  priorityIndicator: {
    width: 8,
    height: 8,
    borderRadius: 4,
    marginRight: 8,
  },
  todoText: {
    fontSize: 16,
  },
  deleteButton: {
    padding: 8,
  },
  emptyState: {
    padding: 24,
    alignItems: "center",
    justifyContent: "center",
  },
  emptyText: {
    textAlign: "center",
    marginBottom: 16,
  },
  emptyButton: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    borderWidth: 1,
    borderRadius: 4,
  },
  emptyButtonText: {
    fontWeight: "500",
  },
});
