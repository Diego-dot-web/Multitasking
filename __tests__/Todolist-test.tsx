import TodoList from "../app/(tabs)/TodoList";
import renderer from "react-test-renderer";

jest.mock("../context/ThemeContext");
// const ThemeContext = require("../context/ThemeContext");
//
// ThemeContext.useTheme = jest.fn(() => {
//   return { isDark: true };
// });

it("renders TodoList component", async () => {
  const tree = await renderer.create(<TodoList />).toJSON();

  expect(tree).toMatchSnapshot();
});
