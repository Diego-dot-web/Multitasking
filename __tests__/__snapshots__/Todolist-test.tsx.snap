// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TodoList component 1`] = `
<View
  style={
    [
      {
        "flex": 1,
        "padding": 16,
      },
      {
        "backgroundColor": "#09090B",
      },
    ]
  }
>
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 4,
          },
          {
            "color": "#FAFAFA",
          },
        ]
      }
    >
      Tus Misiones
    </Text>
    <Text
      style={
        [
          {
            "fontSize": 16,
          },
          {
            "color": "#71717A",
          },
        ]
      }
    >
      Gestiona tus tareas y conquista tus objetivos.
    </Text>
  </View>
  <View
    style={
      [
        {
          "borderRadius": 8,
          "borderWidth": 1,
          "elevation": 2,
          "padding": 16,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 4,
        },
        {
          "backgroundColor": "#18181B",
          "borderColor": "#27272A",
        },
        {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 16,
        }
      }
    >
      <View>
        <Text
          style={
            [
              {
                "fontSize": 18,
                "fontWeight": "bold",
              },
              {
                "color": "#FAFAFA",
              },
            ]
          }
        >
          Lista de Misiones
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 14,
              },
              {
                "color": "#71717A",
              },
            ]
          }
        >
          0
           de 
          0
           misiones completadas (
          0
          %)
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 8,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "#27272A",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 32,
              "justifyContent": "center",
              "opacity": 1,
              "width": 32,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={16}
            bbWidth={16}
            fill="none"
            focusable={false}
            height={16}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="#71717A"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={16}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4285624698,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="2"
            >
              <RNSVGPath
                d="M22 3H2l8 9.46V19l4 2v-8.54L22 3z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "#27272A",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 32,
              "justifyContent": "center",
              "opacity": 1,
              "width": 32,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={16}
            bbWidth={16}
            fill="none"
            focusable={false}
            height={16}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="#71717A"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 16,
                  "width": 16,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={16}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4285624698,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="2"
            >
              <RNSVGPath
                d="M11 5h10"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M11 9h7"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M11 13h4"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M3 17l3 3 3-3"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M6 18V4"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 8,
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Añadir nueva misión..."
          placeholderTextColor="#71717A"
          style={
            [
              {
                "borderRadius": 4,
                "borderWidth": 1,
                "flex": 1,
                "height": 44,
                "paddingHorizontal": 12,
              },
              {
                "backgroundColor": "#18181B",
                "borderColor": "#27272A",
                "color": "#FAFAFA",
              },
            ]
          }
          value=""
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F59E0B20",
              "borderRadius": 4,
              "height": 44,
              "justifyContent": "center",
              "opacity": 1,
              "width": 44,
            }
          }
        >
          <View
            style={
              [
                {
                  "borderRadius": 6,
                  "height": 12,
                  "width": 12,
                },
                {
                  "backgroundColor": "#F59E0B",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9333EA",
            "borderRadius": 4,
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 16,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          fill="none"
          focusable={false}
          height={20}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#fff"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={20}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          >
            <RNSVGCircle
              cx="12"
              cy="12"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              r="10"
            />
            <RNSVGLine
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              x1="12"
              x2="12"
              y1="8"
              y2="16"
            />
            <RNSVGLine
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              x1="8"
              x2="16"
              y1="12"
              y2="12"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          style={
            {
              "color": "#fff",
              "fontWeight": "500",
              "marginLeft": 8,
            }
          }
        >
          Añadir
        </Text>
      </View>
    </View>
    <RCTScrollView
      style={
        [
          {
            "borderRadius": 4,
            "borderWidth": 1,
            "flex": 1,
          },
          {
            "borderColor": "#27272A",
          },
        ]
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "marginBottom": 16,
                  "textAlign": "center",
                },
                {
                  "color": "#71717A",
                },
              ]
            }
          >
            ¡Añade una nueva misión para comenzar tu aventura!
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
